src/metabase/api/setup.clj

```diff
@@ -93,6 +93,9 @@
     ;; notify that we've got a new user in the system AND that this user logged in
     (events/publish-event! :user-create {:user_id (u/get-id new-user)})
     (events/publish-event! :user-login {:user_id (u/get-id new-user), :session_id (str session-id), :first_login true})
+    ; set `source-address-header` to "X-Forwarded-For" if such a header is present
+    (when (-> request :headers (get "x-forwarded-for"))
+      (public-settings/source-address-header "X-Forwarded-For"))
```

Comment:
If headers always come back from ring lowercased, then this looks like it might not do what you expect the `source-address` function.

If I were you I would give `public-settings/source-address-header` a magic get that lowercases the value (hopefully in an English locale 😉) as it comes out so it doesn't matter if you pass in an uppercase value. The docstring for `defsetting` explains how to do that. That way a user doesn't need to know the value of the setting has to be lowercased if they set it themselves, e.g. via the `MB_SOURCE_ADDRESS_HEADER` env var